analysis/bulk-deconvolution/exploratory-notebooks/explore-epic-results.Rmd

```diff
@@ -0,0 +1,401 @@
+compare_reference_celltype <- function(celltype, epic_df) {
+  epic_other_df <- epic_df |>
+    dplyr::filter(epic_celltype == celltype) |>
+    dplyr::select(project_diagnosis, sample_id, reference, fraction) |>
+    dplyr::group_by(project_diagnosis, sample_id) |>
+    tidyr::pivot_wider(
+      names_from = reference, 
+      values_from = fraction
+    ) |>
+    dplyr::mutate(higher_value = ifelse(BRef > TRef, "BRef", "TRef"))
+  
+  ggplot(epic_other_df) + 
+    aes(x = TRef, y = BRef, color = higher_value) + 
+    geom_point(alpha = 0.5) + 
+    geom_abline() +
+    labs(
+      x = "TRef fraction",
+      y = "BRef fraction", 
+      title = celltype
+    ) +
+    facet_wrap(vars(project_diagnosis), nrow = 1)
+}
+
+shared_celltypes |>
+  purrr::map(compare_reference_celltype, epic_df) |>
+  patchwork::wrap_plots(nrow = length(shared_celltypes), guides = "collect")
+```
+From all above comparisons, we see these trends:
+
+* `Bcell`, `CD4_Tcells`,  and `NKcell` proportions are roughly similar between references and low across all samples
+  * In a handful of CNS tumor samples, `TRef` estimates higher `BCell` proportions
+  * In a handful of CNS tumor samples, `TRef` estimates higher `BCell` proportions
+  * It does look like `CD4_Tcells` estimates are a bit higher for `TRef`, but there are two osteosarcoma samples which show really high values from `BRef`
+* `CD8_Tcells` are mostly similar with some additional trends:
+  * Vaues are generally higher for `BRef` for Wilms Tumor samples
+  * Values are generally higher for `TRef` and both CNS tumor and osteosarcoma samples
+* As stated, `TRef` classifies more cells than does `BRef` for all but one osteosarcoma sample
+
+
+**Based on these comparisons, we'll move forward with the `TRef` reference.**
+
+```{r}
+tref_df <- epic_df |>
+  dplyr::filter(reference == "TRef")
+```
+
+
+Let's look at the samples had very few "other" cells:
+
+```{r}
+classified_samples <- tref_df |>
+  dplyr::filter(is_other, fraction <= 0.1) |> 
+  dplyr::pull(sample_id)
+
+classified_samples_df <- tref_df |>
+  dplyr::filter(sample_id %in% classified_samples) |>
+  # add more detailed diagnosis for the non-osteo sample
+  dplyr::mutate(sample_diagnosis = ifelse(
+    project_id == "SCPCP000009", 
+    glue::glue("{sample_id}\n({diagnosis})"), 
+    sample_id)
+  ) |>
+  dplyr::select(
+    project_diagnosis, 
+    sample_diagnosis, 
+    epic_celltype, 
+    fraction) 
+
+ggplot(classified_samples_df) + 
+  aes(x = sample_diagnosis, y = fraction, fill = epic_celltype) +
+  geom_col() + 
+  scale_fill_manual(values = color_map, na.value = "grey90", name = "cell type") + 
+  theme(axis.text.x = element_text(angle = 30, hjust = 1)) + 
+  facet_wrap(vars(project_diagnosis), scales = "free_x")
+```
+* It does make sense that `CAFs` are dominating osteosarcoma samples (indeed they are prevalent, generally speaking, across a few more than these osteosarcoma samples) since we know they are a major component of osteosarcomas
```

Comment:
I think that a very very large proportion of these assignments are incorrect. CAFs and osteo cells are going to be hard to classify because I would assume the genes being used to define a "CAF" is basically the same as what defines a sarcoma cell. To define CAFs vs. tumor here we need to be more specific than the Tref reference probably is. That being said, I think the Bref actually does a better job for these samples than Tref. I would expect to see monocytes in osteo samples and with Bref you actually are identifying "other", which we assume are tumor cells. If we move forward with EPIC I don't think we need to choose one reference for all projects. The cell types in a glioma will not be the same as cell types in a sarcoma.